lib/segment/src/index/field_index/map_index/mmap_map_index.rs

```diff
@@ -198,11 +198,12 @@ impl<N: MapIndexKey + Key + ?Sized> MmapMapIndex<N> {
     }
 
     pub fn get_count_for_value(&self, value: &N) -> Option<usize> {
-        self.value_to_points
-            .get(value)
-            .ok()
-            .flatten()
-            .map(|p| p.len())
+        self.value_to_points.get(value).ok().flatten().map(|ids| {
+            ids.iter()
+                .filter(|idx| !(self.deleted.get(**idx as usize).unwrap_or(false)))
+                .unique()
```

Comment:
Is there a possibility that we actually store non-unique point-id for a value?

Reply:
The tests were failing without this, so I assume yes.

Reply:
not sure how this is possible, maybe we should unique before building hashmap